src/solver/mod.rs

```diff
@@ -29,6 +29,7 @@ pub mod record {
     pub mod mur;
     pub mod sur;
     pub mod rem;
+    pub mod surcc;
```

Comment:
Good job doing this. 

The record implementations are getting out of hand in terms of repeated code, but this is a good solution for now.

Let's start thinking about how we can refactor the `record` module in the future.